src/vtbackend/Grid.cpp

```diff
@@ -391,15 +391,19 @@ CRISPY_REQUIRES(CellConcept<Cell>)
 LineCount Grid<Cell>::scrollUp(LineCount linesCountToScrollUp, GraphicsAttributes defaultAttributes) noexcept
 {
     verifyState();
+    // Number of lines in the ring buffer that are not yet
+    // used by the grid system.
+    auto const linesAvailable = LineCount::cast_from(_lines.size() - unbox<size_t>(_linesUsed));
+    if (std::get_if<Infinite>(&_historyLimit) && linesAvailable < linesCountToScrollUp)
+    {
+        auto const linesToAllocate = linesCountToScrollUp - linesAvailable;
+        for ([[maybe_unused]] auto const _: ranges::views::iota(0, unbox<int>(linesToAllocate)))
```

Comment:
Certainly possible, not sure if worth it. Changing the compiler so that variables following a specific naming pattern, like `_`, or `Unused` are ignored by `-Wunused` sounds more reasonable.

Is it not possible to just simply omit the variable name here.

You just need to use a better API that repeats the invocation of a function X times instead of this kludge.

Reply:
I changed it to use resize, instead so don't need any more for loop.

Reply:
@Utkarsh-khambra fun side note: We're having `crispy::times(N)` that allows you to write code such as:

```cpp
#include <crispy/times.h>
#include <fmt/format.h>
using namespace crispy;
void demo() {
    times(5) | []() { fmt::print("Hello, World\n"); };
    // but also
    times(5) | [](auto i) { fmt::print("Attempt #{}\n", i); }
}
```

Reply:
The `|` operator for times gives a compiler error. :(. I reverted the resize change since it looks like resize at least on libstdc++ only allocates the amount specified in the function argument. So if we would increase our lines in ring buffer by one it will do allocation on each increment.